base_substate/models/base_substate.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2019 Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class BaseSubstateType(models.Model):
+    _name = "base.substate.type"
+    _description = 'Base Substate Type'
+    _order = 'name asc, model asc'
+
+    # Data in this object should be created by import as technical data
+    # in specific module ex : sale_subsatate
+    name = fields.Char('Substate Name', required=True, translate=True)
+    model = fields.Selection(selection=[], string='Apply on', required=True)
```

Comment:
if you don't provide any option this should be a Char field.
Otherwise, as it's a tech field, I'd make `target_state_field` `target_state_field_id` and relate `ir.model.field` then `model` becomes related to the field's model.

Reply:
Thanks for review.
I use selection because this module does not provide a functionality by itself but an abstract model to implement sub state by other (end) module (ex: sale_substate ). So selection limit the choice of model managed by the end module (ex : sale.order). If we use link to ir.model.field or ir.model, user can choose a model that is not managed by the end module. Moreover, data of this object is a technical data created once by importation at install of the end module.

Reply:
ping @simahawk